### PR TITLE
Add new API inner&outer

### PR DESCRIPTION
**Pull request by zhiboniu**
### PR types

New features
### PR changes

APIs
### Describe

Add new API inner&outer

paddle.inner Examples:
```
            import paddle
            x = paddle.arange(1, 7).reshape((2, 3)).astype('float32')
            y = paddle.arange(1, 10).reshape((3, 3)).astype('float32')
            out = paddle.inner(x, y)
            print(out)
            #        ([[14, 32, 50],
            #         [32, 77, 122]])
```

paddle.outer Examples:
```
            import paddle
            x = paddle.arange(1, 4).astype('float32')
            y = paddle.arange(1, 6).astype('float32')
            out = paddle.outer(x, y)
            print(out)
            #        ([[1, 2, 3, 4, 5],
            #         [2, 4, 6, 8, 10],
            #         [3, 6, 9, 12, 15]])
```

### CODE REVIEW

File: python/paddle/tensor/math.py [hunk 1]
```diff
@@ -1070,6 +1070,127 @@ def addmm(input, x, y, beta=1.0, alpha=1.0, name=None):
...
+        ny = y.reshape((-1, yshape[-1]))
+
+        if in_dygraph_mode():
+            return _C_ops.matmul_v2(nx, ny.T).reshape(dstshape)
+
+        def __check_input(x, y):
+            var_names = {'x': x, 'y': y}
+            for name, val in var_names.items():
+                check_variable_and_dtype(val, name,
+                                        ['float16', 'float32', 'float64'], 'inner')
```
Comment (jeff41404) [inline python/paddle/tensor/math.py:1246]: shall we support data type of complexfloat64 and complexfloat128 ?

File: python/paddle/tensor/math.py [hunk 2]
```diff
@@ -1070,6 +1070,127 @@ def addmm(input, x, y, beta=1.0, alpha=1.0, name=None):
...
+    ny = y.reshape((1, -1))
+
+    if in_dygraph_mode():
+        return _C_ops.matmul_v2(nx, ny)
+
+    def __check_input(x, y):
+        var_names = {'x': x, 'y': y}
+        for name, val in var_names.items():
+            check_variable_and_dtype(val, name,
+                                     ['float16', 'float32', 'float64'], 'inner')
```
Comment (jeff41404) [inline python/paddle/tensor/math.py:1309]: shall we support data type of complexfloat64 and complexfloat128 ?

**Review by phlrain — DISMISSED** [review]

File: python/paddle/tensor/math.py (hunk 1 above)
Comment (zhiboniu) [reply python/paddle/tensor/math.py:1246]: 复数后面再支持

File: python/paddle/tensor/math.py (hunk 2 above)
Comment (zhiboniu) [reply python/paddle/tensor/math.py:1309]: 复数后面再支持

**Review by phlrain — DISMISSED** [review]

**Review by jeff41404 — DISMISSED** [review]
LGTM

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: python/paddle/tensor/math.py [hunk 3]
```diff
@@ -1195,6 +1195,127 @@ def addmm(input, x, y, beta=1.0, alpha=1.0, name=None):
...
+        return out.reshape(dstshape)
+
+
+def outer(x, y, name=None):
+    """
+
+    Outer product of two Tensors.
+
+    Args:
+        x (Tensor): The input tensor which is a Tensor.
```
Comment (jzhang533) [inline python/paddle/tensor/math.py]: 这个描述，更新一下吧。

File: python/paddle/tensor/math.py [hunk 4]
```diff
@@ -1195,6 +1195,127 @@ def addmm(input, x, y, beta=1.0, alpha=1.0, name=None):
...
+        y (Tensor): The input tensor which is a Tensor.
+        name(str, optional): The default value is None. Normally there is no need for
+            user to set this property. For more information, please refer to :ref:`api_guide_Name`
+
+    Returns:
+        Tensor: The inner product Tensor.
```
Comment (jzhang533) [inline python/paddle/tensor/math.py]: 写错了。

**Review by jzhang533 — DISMISSED** [review]
lgtm

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by TCChenlong — APPROVED** [review]
lgtm

**Pull request merged**